python/lsst/jointcal/jointcal.py

```diff
@@ -668,12 +678,26 @@ def _fit_astrometry(self, dataName=None):
 
         return Astrometry(fit, model, sky_to_tan_projection)
 
+    def check_stars(self):
+        """Count measured and reference stars per ccd and warn/log them."""
+        for ccdImage in self.associations.getCcdImageList():
+            stars = ccdImage.countStars()
+            self.log.debug("ccdImage %s has %s measured and %s reference stars",
+                           ccdImage.getName(), stars[0], stars[1])
+            if stars[0] < self.config.minMeasuredStarsPerCcd:
+                self.log.warn("ccdImage %s has only %s measuredStars (desired %s)",
+                              ccdImage.getName(), stars[0], self.config.minMeasuredStarsPerCcd)
+            if stars[1] < self.config.minRefStarsPerCcd:
+                self.log.warn("ccdImage %s has only %s RefStars (desired %s)",
+                              ccdImage.getName(), stars[1], self.config.minRefStarsPerCcd)
+
```

Comment:
Can you do more than just log a warning if a count is too low? Is there a threshold for either value below which the fit should give up (e.g. one or two additional config parameters)?

Reply:
That is a very good question. I think I'd rather start by issuing warnings now, and turn it into a `raise` or equivalent after we've gotten a sense of how often it triggers. Right now, I just want to know about it as part of debugging things, and I don't really have a good sense for what an appropriately "safe" value is a priori. @TallJimbo might have opinions there, though?

Reply:
It's probably possible to come up with a minimum for at least some specific models from first principles, but I think it's probably most productive to just log things for now.